CDAP-8036 add a fat coprocessor jar to hbase compat packages

albertshau opened the pull request:
Added a fat jar containing all the CDAP coprocessors to every
hbase compat package in a new 'coprocessor' directory. It is
meant to be used in CDAP setups where the replication status
tool is required, or setups where 'master.manage.hbase.coprocessors'
is set to false in order to enable rolling hbase upgrades.

albertshau committed:
CDAP-8036 add a fat coprocessor jar to hbase compat packages

Added a fat jar containing all the CDAP coprocessors to every
hbase compat package in a new 'coprocessor' directory. It is
meant to be used in CDAP setups where the replication status
tool is required, or setups where 'master.manage.hbase.coprocessors'
is set to false in order to enable rolling hbase upgrades.

sagarkapare commented:
I tried running the command from cdap directory
`MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn clean package -DskipTests -P coprocessors`

Seems like there are 2 issues - 
1. Empty `opt/cdap/${package.name}/coprocessor` directory gets created in all modules. 
2. fat jar is created in the hbase-compat modules, but the path has `${package.name}` in it. seems like variable is not getting resolved.

albertshau commented:
hm good points.  I tested this with an actual rpm, guess something in another profile sets package.name. As for the coprocessor directory in all other modules, was trying to avoid copying this logic in all compat poms, but may have to do that to fix it.

albertshau committed:
CDAP-8036 move everything to individual compat dist profile

The common profile was causing too many issues, so just moved
the two additional executions into every compat module's dist
profile. This will also make it so that existing builds don't
need to change.

sagarkapare commented:
LGTM 👍
